tsdb/head.go

```diff
@@ -1673,25 +1818,34 @@ func (s *memSeries) maxTime() int64 {
 	return c.maxTime
 }
 
-func (s *memSeries) cut(mint int64) *memChunk {
-	c := &memChunk{
+func (s *memSeries) cut(mint int64, chunkReadWriter *chunks.HeadReadWriter) *memChunk {
+	if s.headChunk != nil {
+		chunkRef, err := chunkReadWriter.WriteChunk(s.ref, s.headChunk.minTime, s.headChunk.maxTime, s.headChunk.chunk)
+		if err != nil {
+			panic(err)
```

Comment:
panic? Can we do better?

Reply:
I forgot to put a TODO here. We should be able to get rid of the panic, I will look into it.

Reply:
There is a panic below too in the same function for the appender. And looking through the flow where this is called, hard fail seems to be the pattern in the ingest path if it fails. I will keep it as panic for now.